File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-13.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int n = nums.size();
        int low = 0;
        int right = n-1;
        int mid = 0;

        while(low<=right){
            mid = low + (right-low)/2;
            if(target==nums[mid]){
                return mid;
            }

            if(nums[low]<=nums[mid]){
                if(target>nums[mid] || target<nums[low]){
                low = mid+1;
                }
                else {
                    right = mid -1;
                }

            } else {
                if( target<nums[mid] || target > nums[right]){
                    right = mid -1;
                }
                else {
                    low = mid + 1;
                }
            }
        }
        return -1;
    }
    };

File: Data Structures & Algorithms/combination-target-sum/submission-13.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []

        def backtrack(curr,start,total):
            if total == target:
                res.append(curr[:])
                return

            for i in range(start,len(nums)):
                if total + nums[i] > target:
                    continue
                
                curr.append(nums[i])
                backtrack(curr,i,total+nums[i])

                curr.pop()
                
        backtrack([],0,0)
        return res
                

        